### PR TITLE
  LDTOOLS-7: Sequence Viewer: Implement a way to highlight and differentiate non-natural residues in the sequence viewer

### PR DESCRIPTION
**Pull request by ritikgarggg**
Jira: [LDTOOLS-7](https://schrodinger.atlassian.net/browse/LDTOOLS-7)
Primary: @pradeepnschrodinger @karry08 

Summary: To highlight non-natural monomers, we have added a new getter, `residueTilePropsGetter`, in `zoomer`, which obtains props to style residue tiles. We use this getter to add bottom borders to monomer tiles. Moreover, we have added a fading effect for residue codes of more than `maxResidueCodeLength`. For this, we have filled the text in residue tiles in parts to have different styling for each part. We have made use of the `measureText` canvas API to determine the horizontal offset of subsequent text parts based on the previously filled text.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -461,8 +461,10 @@ zoomer: {
     // canvas
     residueFont: "13px Helvetica Neue",
     residueFontOffset: [0, 0], // horizontal and vertical offset of residue font in px
+    maxResidueCodeLength: 6, // maximum length of residue code to be displayed in the viewer, codes with greater length are truncated with a fading effect 
     canvasEventScale: 1,
-    residueFontGetter: undefined // this can be overriden with a custom function of the following signature: (char: string, pos: {x: number, y: number}) => { font: string, color: string}
+    residueFontPropsGetter: undefined // this can be overriden with a custom function of the following signature: (residueCode: string, pos: {x: number, y: number}) => { font: string, color: string}
+    residueTilePropsGetter: undefined // function to get the properties for a specific residue tile, which can be overriden with a custom function of the following signature: (pos: {x: number, y: number}) => { hasBottomBorder: boolean }
```
Comment (ritikgarggg) [inline README.md]: Should I remove this option, which I exposed for controlling tile characteristics, as we no longer require to modify them as per the new Product/UX requirements (no need of a bottom bar for non-natural monomers)?

Comment (karry08) [reply README.md]: yeah we should remove it if the requirement is final

**Comment by karry08** [comment]
lgtm with comments addressed

File: README.md [hunk 2]
```diff
@@ -461,9 +461,10 @@ zoomer: {
     // canvas
     residueFont: "13px Helvetica Neue",
     residueFontOffset: [0, 0], // horizontal and vertical offset of residue font in px
+    maxResidueCodeLength: 6, // maximum length of residue code to be displayed in the viewer, codes with greater length are truncated with a fading effect 
     canvasEventScale: 1,
-    residueFontGetter: undefined // this can be overriden with a custom function of the following signature: (char: string, pos: {x: number, y: number}) => { font: string, color: string}
-
+    residueFontPropsGetter: undefined // this can be overriden with a custom function of the following signature: (residueCode: string, pos: {x: number, y: number}) => { font: string, color: string}
```
Comment (pradeepnschrodinger) [inline README.md]: What's the reason for the name change?

**Pull request merged**